Improvements

vvaltia opened the pull request.

Commit:
Dont create postgres service is postgres.create if false

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Always create postgres volume

 - If postgres.pvc is false, create emptyDir volume

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Add option for creating "keys" volume in rems

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Add tls option to rems ingress

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Add port to init config

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Add tmp volume if readOnlyFilesystem is true

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Edit values.yaml comments

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Update README

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Improvements

 - Support for postgres and read only filesystems
 - Disable database-wait by default, takes too long
 - Possibility to add db-connection tls certs
 - Add option for image registry

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

Commit:
Add db-tls configmap template

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>

blankdots commented on templates/postgres.yaml:
```diff
@@ -16,9 +16,12 @@ spec:
       containers:
         - name: {{ .Values.postgres.name | default "postgres-rems" }}
-          image: {{ .Values.postgres.image | default "centos/postgresql-96-centos7" }}
+          image: {{ include "imageRegistry" . }}{{ .Values.postgres.image | default "centos/postgresql-96-centos7" }}
```
shall we use the postgres 13 version?

blankdots commented on templates/rems.yaml:
```diff
@@ -19,16 +19,16 @@ spec:
       {{- if .Values.dbConnection.wait }}
       - name: database-wait
-        image: jbergknoff/postgresql-client
+        image: {{ include "imageRegistry" . }}jbergknoff/postgresql-client
```
shall we use the official postgresql image or https://hub.docker.com/r/bitnami/postgresql or we can keep this as well, thinking we can make use of one image instead of using multiple?

blankdots reviewed (approved):
small comments. this looks good

Commit:
Small changes

 - Fix database wait init container command
 - Change used postgres to postgres:13 at dockerhub
 - Hardcode postgres image. Other images might not
   work out of the box

Signed-off-by: Valtteri Valtia <vvaltia@csc.fi>
